### PR TITLE
feat(apm): Fuzzy search on spans

### PR DESCRIPTION
**Pull request by dashed**
- I had to upgrade fuse.js to make travis happy. 

## TODO

- [x] update to designs

### CODE REVIEW

File: src/sentry/static/sentry/app/components/events/interfaces/spans/index.tsx [hunk 1]
```diff
@@ -12,22 +13,46 @@ type PropType = {
...
-          <TraceView event={event} />
-        </PanelBody>
-      </Panel>
+      <div>
+        <SearchBar
+          defaultQuery=""
+          query={this.state.searchQuery || ''}
+          placeholder={t('Filter on spans')}
+          onSearch={this.handleSpanFilter}
+        />
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/spans/index.tsx]: In other parts of the application the panel header contains the search controls. For example Settings > Members

File: src/sentry/static/sentry/app/components/events/interfaces/spans/spanTree.tsx [hunk 2]
```diff
@@ -38,13 +42,69 @@ class SpanTree extends React.Component<PropType> {
...
+    } = input;
+
+    const messages: React.ReactNode[] = [];
+
+    const showHiddenSpansMessage = !isCurrentSpanHidden && numOfSpansOutOfViewAbove > 0;
+
+    if (showHiddenSpansMessage) {
+      messages.push(
+        <span key="spans-out-of-view">
+          <strong>{t('Spans out of view:')}</strong> {numOfSpansOutOfViewAbove}
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/spans/spanTree.tsx]: Should this be more natural language like `%d spans out of view`?

File: src/sentry/static/sentry/app/components/events/interfaces/spans/spanTree.tsx [hunk 3]
```diff
@@ -38,13 +42,69 @@ class SpanTree extends React.Component<PropType> {
...
+    }
+
+    const showFilteredSpansMessage =
+      !isCurrentSpanFilteredOut && numOfFilteredSpansAbove > 0;
+
+    if (showFilteredSpansMessage) {
+      if (!isCurrentSpanHidden) {
+        messages.push(
+          <span key="spans-filtered">
+            <strong>{t('Spans not matching search:')}</strong> {numOfFilteredSpansAbove}
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/events/interfaces/spans/spanTree.tsx]: Same here `%d spans filtered out`

Comment (dashed) [reply src/sentry/static/sentry/app/components/events/interfaces/spans/spanTree.tsx]: done.

File: src/sentry/static/sentry/app/components/events/interfaces/spans/spanTree.tsx (hunk 2 above)
Comment (dashed) [reply src/sentry/static/sentry/app/components/events/interfaces/spans/spanTree.tsx]: done.

File: src/sentry/static/sentry/app/components/events/interfaces/spans/index.tsx (hunk 1 above)
Comment (dashed) [reply src/sentry/static/sentry/app/components/events/interfaces/spans/index.tsx]: @markstory looks like this is inline with the latest designs for the trace view on Figma. @doralchan will be killing the panel

**Comment by dashed** [comment]
I had to upgrade fuse.js to make travis happy.

**Review by doralchan — APPROVED** [review]

**Pull request merged**